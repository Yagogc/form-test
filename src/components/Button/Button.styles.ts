import styled, { css } from 'styled-components/macro'

const ButtonGroup = styled.div`
  ${() => css`
    margin-top: 20px;
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;
  `}
`

const ButtonContainer = styled.button`
  ${({ back, disabled }: { back: boolean; disabled: boolean }) => css`
    padding: 5px 10px;
    background-color: ${back ? 'lightgrey' : 'green'};
    font-size: 20px;
    color: ${back ? 'inherit' : 'white'};
    border: 0;
    box-shadow: none;
    border-radius: 5px;
    cursor: ${disabled ? 'not-allowed' : 'pointer'};
    transition: all 0.2s ease-in-out;
    opacity: ${disabled ? '0.5' : '1'};

    &:hover {
      transform: scale(0.9);
    }
    &:active {
      transform: scale(1);
    }
  `}
`

export { ButtonGroup, ButtonContainer }
