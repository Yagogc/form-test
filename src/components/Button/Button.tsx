import React from 'react'
import { ButtonContainer } from './Button.styles'

interface ButtonInterface {
  type: 'button' | 'submit' | 'reset'
  back?: boolean
  onClick?: Function
}

const Button: React.FC<ButtonInterface> = ({
  children,
  type,
  back = false,
  onClick = () => null,
}) => {
  return (
    <ButtonContainer type={type} back={back} onClick={() => onClick()}>
      {children}
    </ButtonContainer>
  )
}

export default Button
