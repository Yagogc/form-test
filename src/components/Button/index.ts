import Button from './Button'
import { ButtonGroup } from './Button.styles'

export { ButtonGroup, Button }
