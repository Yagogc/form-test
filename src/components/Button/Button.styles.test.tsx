import React from 'react'
import { render } from '@testing-library/react'
import { ButtonContainer } from './Button.styles'
import 'jest-styled-components'

describe('ButtonContainer', () => {
  it('has default styles', () => {
    const { container } = render(<ButtonContainer />)

    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyleRule('transform', 'scale(0.9)', {
      modifier: ':hover',
    })
    expect(container.firstChild).toHaveStyleRule('transform', 'scale(1)', {
      modifier: ':active',
    })
  })

  it('should render diferents styles with "back" prop', () => {
    const { container, rerender } = render(<ButtonContainer />)

    expect(container.firstChild).toHaveStyleRule('background-color', 'green')

    rerender(<ButtonContainer back />)

    expect(container.firstChild).toHaveStyleRule(
      'background-color',
      'lightgrey'
    )
  })

  it('should render diferents styles with "back" prop', () => {
    const { container, rerender } = render(<ButtonContainer />)

    expect(container.firstChild).toHaveStyleRule('cursor', 'pointer')
    expect(container.firstChild).toHaveStyleRule('opacity', '1')

    rerender(<ButtonContainer disabled />)

    expect(container.firstChild).toHaveStyleRule('cursor', 'not-allowed')
    expect(container.firstChild).toHaveStyleRule('opacity', '0.5')
  })
})
