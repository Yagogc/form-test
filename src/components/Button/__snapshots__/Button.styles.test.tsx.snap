// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer has default styles 1`] = `
.c0 {
  padding: 5px 10px;
  background-color: green;
  font-size: 20px;
  color: white;
  border: 0;
  box-shadow: none;
  border-radius: 5px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  opacity: 1;
}

.c0:hover {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

.c0:active {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

<button
  class="c0"
/>
`;
